### PR TITLE
Loop driver, NBD and install instructions

### PR DESCRIPTION
**Pull request by JoeReid**
Added updated module options for the loop driver and NBD module

Changed the layout of the install section slightly

Started a new troubleshooting page (WIP) for common issues
(Just gathering error messages etc now)

Changed invalid CLI page location

### CODE REVIEW

**Comment by JoeReid** [comment]
I have updated the troubleshooting page with some basic information.
I am removing the WIP status now.

File: _docs/install/prerequisites/devicepresentation.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+module: install/prerequisites/devicepresentaion.md
+---
+
+# Device presentation
+
+StorageOS requires one of two runtime requirements to present virtual block devices
+from Userspace. While not mandatory, the NBD aproach does offer significantly
+better performance over the loop-device alternative.
+
+## Loopback driver
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: I suggest putting NBD above this section, and renaming Loopback driver to Loopback driver (fallback). Or numbering these alternatives as 1. and 2. to make it clear that you only need to do one.

File: _docs/install/prerequisites/devicepresentation.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+layout: guide
+title: StorageOS Docs - Device Presentation
+anchor: install
+module: install/prerequisites/devicepresentaion.md
+---
+
+# Device presentation
+
+StorageOS requires one of two runtime requirements to present virtual block devices
+from Userspace. While not mandatory, the NBD aproach does offer significantly
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: Userspace => userspace

File: _docs/install/prerequisites/devicepresentation.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+1. Add the following module configuration lines in `/etc/modprobe.d/nbd.conf`
+
+    ```text
+    options nbd nbds_max=1024
+    options nbd max_part=15
+    ```
+
+Your distribution may set different defaults for the provided
+parameters. For storageos to run correctly, our values must be used in their
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: storageos => StorageOS.

File: _docs/install/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+## Common issues
+
+```
+Error response from daemon: dial unix /run/docker/plugins/a-very-long-hash-value/storageos.sock: connect: connection refused
+```
+
+This error indicates that the StorageOS plugin container did not successfully
+start during installation. This is usualy due to incorrect option values
```
Comment (oicheryl) [inline _docs/install/troubleshooting.md]: usally => usually

File: _guides/002-install_and_run.md [hunk 5]
```diff
@@ -14,17 +14,34 @@ sections:
...
+      -
+        title: Docker Volume plugin
+        description: Docker managed plugin install for Docker Engine 1.13+
+        module: install/docker/plugin
+      -
+        title: Docker Swarm
+        description: Installing on Docker Swarm
+        module: install/schedulers/dockerswarm
   -
     title: Schedulers
```
Comment (oicheryl) [inline _guides/002-install_and_run.md]: We should bring the Kubernetes and Docker Swarm docs up a level, then remove this Schedulers section entirely.

**Comment by JoeReid** [comment]
@croomes @tobysmith @oicheryl Requested changes have been made

**Pull request merged**